biomodels/src/main/java/org/reactome/release/ModelsTSVParser.java

```diff
@@ -0,0 +1,65 @@
+package org.reactome.release;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModelsTSVParser {
+    private static final Logger logger = LogManager.getLogger();
+
+    private ModelsTSVParser() { }
+
+    /**
+     * Parses the contents of the models2pathways.tsv file, returning a Map of ReactomePathwayIds=[BioModelsIdentifiers,...]
+     * @param tsvFile -- String, path/to/models2pathways.tsv
+     * @return Map of ReactomePathwayIds and a List of BioModelsIdentifiers
+     */
+    public static Map<String, List<String>> parse(String tsvFile) {
+        Map<String, List<String>> pathwayToBiomodelsIds = new HashMap<>();
+
+        if (tsvFile == null || tsvFile.isEmpty()) {
+            return pathwayToBiomodelsIds;
+        }
+
+        try(BufferedReader br = new BufferedReader(new FileReader(tsvFile))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] fields = line.split("\t");
+                String biomodelsId = fields[0];
+                String pathwayStableId = fields[1];
+
+                if (!matchBioModelsPattern(biomodelsId) || !matchStableIdPattern(pathwayStableId)) {
+                    logger.warn("Line has improperly formatted BioModel ID or Stable ID -- skipping");
```

Comment:
Maybe test the conditions separately so the user knows exactly which one was the problem?